### PR TITLE
rn-001-types-declaration

### PR DESCRIPTION
**Pull request by CiastowskiMichal**

### CODE REVIEW

File: src/components/MainScreen/MainScreen.tsx [hunk 1]
```diff
@@ -43,26 +40,26 @@ const MainScreen = () => {
...
+      const goal : Goal = {
         id: uuidv4(),
         name: text
       };
       return prevGoals.concat(goal);
     });
   };
 
-  const DeleteGoalHandler = (item: any) => {
+  const DeleteGoalHandler = (item: Goal) => {
```
Comment (Stared55) [inline src/components/MainScreen/MainScreen.tsx:52]: lowercase

File: src/components/Item/Item.tsx [hunk 2]
```diff
@@ -5,9 +5,9 @@ import {
 import ClearButton from "../ClearButton/ClearButton";
 import item from './Item.styles';
 import MainScreen from "../MainScreen/MainScreen"
+import { Goal } from "../../types/goal";
 
-interface ItemProps { item: any, onDeleteGoal : (goalId : string) => void }
-
+interface ItemProps { item: Goal, onDeleteGoal : (goal : Goal) => void }
```
Comment (Stared55) [inline src/components/Item/Item.tsx:10]:
```
interface ItemPropsType { 
  item: Goal, 
  onDeleteGoal : (goal : Goal) => void 
}
```

File: src/components/Item/Item.tsx [hunk 3]
```diff
@@ -5,9 +5,9 @@ import {
 import ClearButton from "../ClearButton/ClearButton";
 import item from './Item.styles';
```
Comment (Stared55) [inline src/components/Item/Item.tsx:6]: import styles from './Item.styles';

File: src/components/MainScreen/MainScreen.tsx [hunk 4]
```diff
@@ -8,15 +8,12 @@ import {
...
-  id: string,
-  name: string
-}
 
 const MainScreen = () => {
 
-  const originalGoals: Aa[] = [{
+  const originalGoals: Goal[] = [{
     id: uuidv4(),
     name: 'Take photos'
```
Comment (Stared55) [inline src/components/MainScreen/MainScreen.tsx:18]: implement i18n https://react.i18next.com/latest/usetranslation-hook

File: src/components/Item/Item.tsx [hunk 5]
```diff
@@ -5,9 +5,9 @@ import {
...
 import MainScreen from "../MainScreen/MainScreen"
+import { Goal } from "../../types/goal";
 
-interface ItemProps { item: any, onDeleteGoal : (goalId : string) => void }
-
+interface ItemProps { item: Goal, onDeleteGoal : (goal : Goal) => void }
 
 const Item = (props: ItemProps) => {
```
Comment (Stared55) [inline src/components/Item/Item.tsx:12]: const Item = ({ onDeleteGoal }: ItemProps) => {

File: src/components/MainScreen/MainScreen.tsx [hunk 6]
```diff
@@ -8,15 +8,12 @@ import {
 import AddGoal from '../AddGoal/AddGoal';
 import Item from '../Item/Item';
 import mainscreen from './MainScreen.styles';
```
Comment (Stared55) [inline src/components/MainScreen/MainScreen.tsx:10]: please install extension for VSC **Code Spell Checker**

File: src/components/MainScreen/MainScreen.tsx [hunk 7]
```diff
@@ -43,26 +40,26 @@ const MainScreen = () => {
...
 
-    setGoals((prevGoals: any) => {
-      const goal = {
+    setGoals((prevGoals: Goal[]) => {
+      const goal : Goal = {
         id: uuidv4(),
         name: text
       };
       return prevGoals.concat(goal);
     });
```
Comment (Stared55) [inline src/components/MainScreen/MainScreen.tsx:49]:
```
    const newGoal: Goal = {
      id: uuidv4(),
      name: text
    }

    setGoals([...goalList, newGoal])
```

File: src/components/ClearButton/ClearButton.tsx [hunk 8]
```diff
@@ -2,17 +2,17 @@ import React from 'react';
 import {
   Pressable, Text
 } from 'react-native';
-import clearbutton from './ClearButton.styles';
+import styles from './ClearButton.styles';
 
-interface ClearButtonProps {onPress : () => void, title : string};
+interface ClearButtonProps { onPress: () => void, title: string };
```
Comment (Stared55) [inline src/components/ClearButton/ClearButton.tsx:7]: We can go with the pattern with keeping the types/interfaces in the separated files, so ->
move the interface ClearButtonProps into -> ClearButton.types.ts

File: src/components/Item/Item.tsx [hunk 9]
```diff
@@ -3,20 +3,22 @@ import {
...
-import item from './Item.styles';
-import MainScreen from "../MainScreen/MainScreen"
+import styles from './Item.styles';
 import { Goal } from "../../types/goal";
 
-interface ItemProps { item: Goal, onDeleteGoal : (goal : Goal) => void }
+interface ItemProps {
+    item: Goal,
+    onDeleteGoal: (goal: Goal) => void
+}
```
Comment (Stared55) [inline src/components/Item/Item.tsx:12]: same

File: src/components/MainScreen/MainScreen.tsx [hunk 10]
```diff
@@ -40,34 +40,29 @@ const MainScreen = () => {
...
-    setGoals((prevGoals: Goal[]) => {
-      const goal : Goal = {
-        id: uuidv4(),
-        name: text
-      };
-      return prevGoals.concat(goal);
-    });
+    const newGoal : Goal = {
+      id: uuidv4(),
+      name: text
```
Comment (Stared55) [inline src/components/MainScreen/MainScreen.tsx:45]: lets rename the param to `name` isntead of `text` if we exactly know what it is

File: src/components/NewButton/NewButton.tsx [hunk 11]
```diff
@@ -2,14 +2,14 @@ import React from 'react';
 import {
   Pressable, Text
 } from 'react-native';
-import newbutton from './NewButton.styles';
+import styles from './NewButton.styles';
 
 interface NewButtonProps {onPress : () => void, title : string};
```
Comment (Stared55) [inline src/components/NewButton/NewButton.tsx:7]: same

**Comment by Stared55** [comment]
and if u fix smth, please add the comment below the my comment (e.g updated)

File: src/components/MainScreen/MainScreen.tsx (hunk 1 above)
Comment (CiastowskiMichal) [reply src/components/MainScreen/MainScreen.tsx]: update

File: src/components/Item/Item.tsx (hunk 2 above)
Comment (CiastowskiMichal) [reply src/components/Item/Item.tsx]: update

File: src/components/Item/Item.tsx (hunk 3 above)
Comment (CiastowskiMichal) [reply src/components/Item/Item.tsx]: update

File: src/components/MainScreen/MainScreen.tsx (hunk 4 above)
Comment (CiastowskiMichal) [reply src/components/MainScreen/MainScreen.tsx:18]: cookie-001

File: src/components/Item/Item.tsx (hunk 5 above)
Comment (CiastowskiMichal) [reply src/components/Item/Item.tsx]: updated

File: src/components/MainScreen/MainScreen.tsx (hunk 7 above)
Comment (CiastowskiMichal) [reply src/components/MainScreen/MainScreen.tsx]: updated

File: src/components/ClearButton/ClearButton.tsx (hunk 8 above)
Comment (CiastowskiMichal) [reply src/components/ClearButton/ClearButton.tsx]: updated

File: src/components/Item/Item.tsx (hunk 9 above)
Comment (CiastowskiMichal) [reply src/components/Item/Item.tsx]: updated

File: src/components/MainScreen/MainScreen.tsx (hunk 10 above)
Comment (CiastowskiMichal) [reply src/components/MainScreen/MainScreen.tsx]: updated

File: src/components/NewButton/NewButton.tsx (hunk 11 above)
Comment (CiastowskiMichal) [reply src/components/NewButton/NewButton.tsx]: updated

File: src/components/AddGoal/AddGoal.tsx [hunk 12]
```diff
@@ -22,7 +20,7 @@ const AddGoal = ({onAddGoal}: AddGoalProps) => {
             <TextInput onChangeText={newText => setEnteredText(newText)}
                 defaultValue={enteredText}
                 style={styles.addGoalInput} />
-            <NewButton onPress={addGoalHandler} title="Add new Goal"></NewButton>
+            <CustomButton onPress={addGoalHandler} title="Add new Goal"></CustomButton>
```
Comment (Stared55) [inline src/components/AddGoal/AddGoal.tsx:23]: after merging https://github.com/CiastowskiMichal/RnToDoTS/pull/4, please make pull and use i18n

File: src/components/AddGoal/AddGoal.types.ts [hunk 13]
```diff
@@ -0,0 +1,5 @@
+interface AddGoalProps {
```
Comment (Stared55) [inline src/components/AddGoal/AddGoal.types.ts:1]: Sounds like a function instead of component

File: src/components/Item/Item.tsx [hunk 14]
```diff
@@ -4,12 +4,7 @@ import {
...
 import styles from './Item.styles';
-import { Goal } from "../../types/goal";
-
-interface ItemProps {
-    item: Goal,
-    onDeleteGoal: (goal: Goal) => void
-}
+import ItemProps from "./Item.types";
 
 const Item = ({item, onDeleteGoal} : ItemProps) => {
```
Comment (Stared55) [inline src/components/Item/Item.tsx:9]: Item of what ?

File: src/components/MainScreen/MainScreen.tsx [hunk 15]
```diff
@@ -35,14 +35,14 @@ const MainScreen = () => {
...
-    if (!text.length) {
+  const addNewGoalHandler = (name: string) => {
+    if (!name.length) {
       return
     }
 
     const newGoal : Goal = {
       id: uuidv4(),
-      name: text
+      name: name
```
Comment (Stared55) [inline src/components/MainScreen/MainScreen.tsx:45]:
```
{
      id: uuidv4(),  
      name,
}
```

Comment (CiastowskiMichal) [reply src/components/MainScreen/MainScreen.tsx]: updated

File: src/components/Item/Item.tsx (hunk 14 above)
Comment (CiastowskiMichal) [reply src/components/Item/Item.tsx]: updated

File: src/components/AddGoal/AddGoal.types.ts (hunk 13 above)
Comment (CiastowskiMichal) [reply src/components/AddGoal/AddGoal.types.ts]: updated

File: src/components/AddGoal/AddGoal.tsx (hunk 12 above)
Comment (CiastowskiMichal) [reply src/components/AddGoal/AddGoal.tsx]: Ok

File: src/components/AddGoalForm/AddGoalForm.styles.tsx [hunk 16]
```diff
@@ -1,7 +1,7 @@
 import React from "react";
 import { StyleSheet } from "react-native";
 
-const addgoal = StyleSheet.create({
+const addgoalform = StyleSheet.create({
```
Comment (Stared55) [inline src/components/AddGoalForm/AddGoalForm.styles.tsx:4]: camelcasel

File: src/components/AddGoalForm/AddGoalForm.tsx [hunk 17]
```diff
@@ -1,10 +1,10 @@
 import React, { useState } from 'react';
 import { TextInput, View } from 'react-native';
 import CustomButton from '../CustomButton/CustomButton';
-import styles from './AddGoal.styles';
-import AddGoalProps from './AddGoal.types';
+import styles from './AddGoalForm.styles';
+import AddGoalFormProps from './AddGoalForm.types';
 
-const AddGoal = ({onAddGoal}: AddGoalProps) => {
+const AddGoalForm = ({onAddGoal}: AddGoalFormProps) => {
```
Comment (Stared55) [inline src/components/AddGoalForm/AddGoalForm.tsx:7]: onAddGoal -> onSubmit ? What do u think ?

**Review by Stared55 — APPROVED** [review]
Fix and merge ✅

**Pull request merged**